libraries/persistent_store/src/format.rs

```diff
@@ -0,0 +1,951 @@
+    pub fn bytes_to_words(&self, bytes: usize) -> usize {
+        div_ceil(bytes, self.word_size())
+    }
+}
+
+/// The word index of the init info in a page.
+const INIT_WORD: usize = 0;
+
+/// The word index of the compact info in a page.
+const COMPACT_WORD: usize = 1;
+
+/// The word index of the content of a page.
+///
+/// Since a page is at least 8 words, there is always at least 6 words of content.
+const CONTENT_WORD: usize = 2;
+
+/// The checksum for a single word.
+///
+/// Since checksums are the number of bits set to zero and a word is 32 bits, we need 5 bits to
+/// store numbers between 0 and 27 (which is 32 - 5).
+const WORD_CHECKSUM: Checksum = Checksum {
+    field: Field { pos: 27, len: 5 },
+};
+
+// The fields of the init info of a page.
+bitfield! {
+    /// The number of times the page has been erased.
+    INIT_CYCLE: Field <= MAX_ERASE_CYCLE,
+
+    /// The word index of the first entry in this virtual page.
+    INIT_PREFIX: Field <= div_ceil(MAX_VALUE_LEN, WORD_SIZE),
+
+    #[cfg(test)]
+    LEN_INIT: Length,
+}
+
+// The fields of the compact info of a page.
+bitfield! {
+    /// The distance in words between head and tail at compaction.
+    ///
+    /// In particular, compaction copies non-deleted user entries from the head to the tail as long
+    /// as entries span the page to be compacted.
+    COMPACT_TAIL: Field <= MAX_VIRT_PAGE_SIZE * MAX_PAGE_INDEX,
+
+    #[cfg(test)]
+    LEN_COMPACT: Length,
+}
+
+// Overview of the first word of the different kind of entries.
+//
+// Each column represents a bit of the word. The first 2 lines give the position in hexadecimal of
+// the bit in the word (the exponent of 2 when the word is written in binary). Each entry starts
+// with the sequence of bits of its identifier. The dots following the identifier are the number of
+// bits necessary to hold the information of the entry (including the checksum). The remaining free
+// bits after the dots are not used by the entry.
+//
+//         0               1
+//         0123456789abcdef0123456789abcdef
+// padding 0
+//  header 10..............................
+//   erase 11000...........
+//   clear 11001.................
+//  marker 11010..........
+//  remove 11011.................
+//
+// NOTE: We could pad the internal entries to the right by extending their identifier. This permits
+// to free some space for shorter identifier for future kind of entries.
+
+// The fields of a padding entry.
+bitfield! {
+    /// The identifier for padding entries.
+    ID_PADDING: ConstField = [0],
+}
+
+// The fields of a user entry.
+bitfield! {
+    /// The identifier for user entries.
+    ID_HEADER: ConstField = [1 0],
+
+    /// Whether the user entry is deleted.
+    HEADER_DELETED: Bit,
+
+    /// Whether the last bit of the user data is flipped.
+    HEADER_FLIPPED: Bit,
+
+    /// The length in bytes of the user data.
+    // NOTE: It is possible to support values of length 1024 by having a separate kind of entries
```

Comment:
Actually 1023 would be the absolute max for an entry. But note that this is the store library, this is not the CTAP storage. You can see entries in this store as blocks in a file system. If CTAP objects (which can be considered as files) do not fit in a single block, we can fragment them as documented [here](https://github.com/ia0/OpenSK/blob/v2/libraries/persistent_store/src/doc.md#fragmentation). I didn't implement it yet because we don't have this blob, and I don't know yet if I'll go with something specific to this blob in CTAP storage, or if I add the fragmentation layer to the store library.